src/cache.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  restoreCache as restoreCacheAction,
+  saveCache as saveCacheAction,
+} from '@actions/cache'
+import {getInput as getInputAction} from '@actions/core'
+import {exec} from '@actions/exec'
+import {arch, homedir} from 'os'
+import {join} from 'path'
+
+type CacheHits = {
+  iso: boolean
+  kic: boolean
+  preload: boolean
+}
+
+export async function restoreCaches(): Promise<CacheHits> {
+  const cacheHits: CacheHits = {iso: true, kic: true, preload: true}
+  if (!useCache()) {
+    return cacheHits
+  }
+  const minikubeVersion = await getMinikubeVersion()
+  const isoCacheKey = restoreCache('iso', minikubeVersion)
+  const kicCacheKey = restoreCache('kic', minikubeVersion)
+  const preloadCacheKey = restoreCache('preloaded-tarball', minikubeVersion)
+  cacheHits.iso = typeof (await isoCacheKey) !== 'undefined'
+  cacheHits.kic = typeof (await kicCacheKey) !== 'undefined'
+  cacheHits.preload = typeof (await preloadCacheKey) !== 'undefined'
+  return cacheHits
+}
+
+async function getMinikubeVersion(): Promise<string> {
```

Comment:
this is not used yet, right ? if so lets comment on top of it (to be used later to cache binary)

Reply:
This is currently being used for adding the version to cache keys.

Reply:
does that mean if Preload stays same but minikube version changes, it wont use that prelaod from old minikube version in cache ?

Reply:
That's correct, my one worry is if we bump the preload version (v18) between releases that we don't want to keep using the old version.

The two options I can think of if we don't want to change the preload cache between minikube versions is:
1. Look at the contents of the preload directory to get the version
2. Add a preload version command to minikube that reports the version of the preload